test/screens/play/play_page_test.dart

```diff
@@ -41,6 +43,54 @@ void main() {
     expect(find.byType(PlayerHand), findsOneWidget);
   });
 
+  testWidgets('should display the game over page with correct text if we have a game result',
+          (WidgetTester tester) async {
+        final state = makeGameState();
+
+        //Test for a win
+        state.endGame(GameResult.WIN);
+
+        await tester
+            .pumpWidget(createWidgetForTesting(child: PlayPage('Play Page', state)));
+        expect(find.byType(GameEndPage), findsOneWidget);
+        expect(find.byType(PlayerHand), findsNothing);
+
+        final gameWinTextFinder = find.text(GameResult.WIN.description);
+        expect(gameWinTextFinder, findsOneWidget);
+
+        final button1Finder = find.text('New Game');
+        final button2Finder = find.text('Exit');
+
+        expect(button1Finder, findsOneWidget);
+        expect(button2Finder, findsOneWidget);
```

Comment:
I'd be tempted to split this up into a bunch of smaller tests. For the `play_page`, all we need are:

`should display the game over page when a result is set`
`should not display the game over page when there is no result set`

And then we'd have a new `game_end_test.dart` which would cover things like:

`should display the right text based on whether the result is win or lose`
`pressing New Game should push to a new Play Page with fresh state`
`pressing Exit should return to the main menu page`

[This post](https://stackoverflow.com/a/51983194/3870219) has some promising looking suggestions for testing the navigation easily :crossed_fingers: